test/index.js

```diff
@@ -187,25 +187,69 @@ describe('index', function() {
       this.plugin.lookupLogicalResourceIds();
       statement.Resource.should.have.members([
         'LogicalResourceId/DDBTable',
-        'LogicalResourceId/S3Bucket2'
+        'LogicalResourceId/AnotherS3Bucket'
       ]);
       should.not.exist(statement.ResourceType);
       should.not.exist(statement.ExcludeResource);
     });
 
-    it("works even if 'Resource' property does not exist in the stack policy statement", function() {
+    it("filters out resources based on 'ExcludeResourcePrefix' in the stack policy statement", function() {
       const statement = {
         Effect: 'Deny',
         Principal: '*',
         Action: [
           'Update:Replace',
           'Update:Delete'
         ],
+        Resource: [
+          'LogicalResourceId/DDBTable'
+        ],
+        ResourceType: [
+          'AWS::S3::Bucket'
+        ],
+        ExcludeResourcePrefix: [
+          'LogicalResourceId/S3Buck'
+        ]
+      };
+
+      this.serverless.service.provider.stackPolicy.push(
+        {
+          Effect: 'Allow',
+          Principal: '*',
+          Action: 'Update:*',
+          Resource: '*'
+        },
+        statement
+      );
+
+      this.plugin.lookupLogicalResourceIds();
+      statement.Resource.should.have.members([
+        'LogicalResourceId/DDBTable',
+        'LogicalResourceId/AnotherS3Bucket'
+      ]);
+      should.not.exist(statement.ResourceType);
+      should.not.exist(statement.ExcludeResource);
```

Comment:
Could we also add

```
should.not.exist(statement.ExcludeResourcePrefix);
```
to the appropriate places? I believe Serverless will halt if it doesn't get deleted as it doesn't match the AWS YAML schema.

Reply:
ebf70760163a1a5fc8bc50cbb4fb08707a298c63